src/server/scripts/EasternKingdoms/Stratholme/boss_baron_rivendare.cpp

```diff
@@ -42,144 +34,119 @@ enum Spells
-                instance->SetData(TYPE_BARON, NOT_STARTED);
-        }
+    void JustDied(Unit* /*killer*/) override
+    {
+        instance->SetData(TYPE_BARON, DONE);
+    }
 
-        void AttackStart(Unit* who) override
-        {
-            //can't use entercombat(), boss' dmg aura sets near players in combat, before entering the room's door
-            if (instance->GetData(TYPE_BARON) == NOT_STARTED)
-                instance->SetData(TYPE_BARON, IN_PROGRESS);
-            ScriptedAI::AttackStart(who);
-        }
+    void UpdateAI(uint32 diff) override
+    {
+        if (!UpdateVictim())
+            return;
 
-        void JustSummoned(Creature* summoned) override
-        {
-            if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0))
-                summoned->AI()->AttackStart(target);
-        }
+        events.Update(diff);
 
-        void JustDied(Unit* /*killer*/) override
-        {
-            instance->SetData(TYPE_BARON, DONE);
-        }
+        if (me->HasUnitState(UNIT_STATE_CASTING))
+            return;
 
-        void UpdateAI(uint32 diff) override
+        while (uint32 eventId = events.ExecuteEvent())
         {
-            if (!UpdateVictim())
-                return;
-
-            //ShadowBolt
-            if (ShadowBolt_Timer <= diff)
+            switch (eventId)
             {
-                if (SelectTarget(SELECT_TARGET_RANDOM, 0))
-                    DoCastVictim(SPELL_SHADOWBOLT);
-
-                ShadowBolt_Timer = 10000;
-            } else ShadowBolt_Timer -= diff;
-
-            //Cleave
-            if (Cleave_Timer <= diff)
-            {
-                DoCastVictim(SPELL_CLEAVE);
-                //13 seconds until we should cast this again
-                Cleave_Timer = 7000 + (rand32() % 10000);
-            } else Cleave_Timer -= diff;
-
-            //MortalStrike
-            if (MortalStrike_Timer <= diff)
-            {
-                DoCastVictim(SPELL_MORTALSTRIKE);
-                MortalStrike_Timer = 10000 + (rand32() % 15000);
-            } else MortalStrike_Timer -= diff;
-
-            //RaiseDead
-            //            if (RaiseDead_Timer <= diff)
-            //          {
-            //      DoCast(me, SPELL_RAISEDEAD);
-            //                RaiseDead_Timer = 45000;
-            //            } else RaiseDead_Timer -= diff;
-
-            //SummonSkeletons
-            if (SummonSkeletons_Timer <= diff)
-            {
-                me->SummonCreature(11197, ADD_POS_1, TEMPSUMMON_TIMED_DESPAWN, 29000);
-                me->SummonCreature(11197, ADD_POS_2, TEMPSUMMON_TIMED_DESPAWN, 29000);
-                me->SummonCreature(11197, ADD_POS_3, TEMPSUMMON_TIMED_DESPAWN, 29000);
-                me->SummonCreature(11197, ADD_POS_4, TEMPSUMMON_TIMED_DESPAWN, 29000);
-                me->SummonCreature(11197, ADD_POS_5, TEMPSUMMON_TIMED_DESPAWN, 29000);
-                me->SummonCreature(11197, ADD_POS_6, TEMPSUMMON_TIMED_DESPAWN, 29000);
```

Comment:
where is this handled now?

Reply:
Currently nowhere, it's missing since I moved the target positions to all spells (via SQL). I'm currently thinking about the implementation how Baron Rivendare should cast the spells to summon the skeletons.